src/react/components/pages/editorPage/canvas.test.tsx

```diff
@@ -219,31 +222,166 @@ describe("Editor Canvas", () => {
             .toMatchObject([MockFactory.createTestRegion("test1"), MockFactory.createTestRegion("test2")]);
     });
 
-    it("Applies tag to selected region", () => {
+    it("onRegionSelected with invalid id throws error", () => {
         const wrapper = createComponent();
-        const onAssetMetadataChanged = jest.fn();
-        wrapper.setProps({ onAssetMetadataChanged });
         const canvas = wrapper.instance();
 
-        canvas.editor.onRegionSelected("test1", null);
-
-        const newTag = MockFactory.createTestTag();
-        canvas.applyTag(newTag.name);
+        expect(() => {
+            canvas.editor.onRegionSelected("fakeRegion", false);
+        }).toThrowError("Couldn't find region with id fakeRegion");
+    });
 
-        const original = getAssetMetadata();
-        const expected: IAssetMetadata = {
+    function cloneWithUpdatedRegionTags(original: IAssetMetadata, regionId: string, tags: string[]) {
+        return {
             ...original,
             regions: original.regions.map((r) => {
-                if (r.id === "test1") {
+                if (r.id === regionId) {
                     return {
                         ...r,
-                        tags: [newTag.name],
+                        tags,
                     };
                 }
                 return r;
             }),
         };
+    }
+
+    it("Applies single tag to selected region", () => {
+        const wrapper = createComponent();
+        const onAssetMetadataChanged = jest.fn();
+        wrapper.setProps({ onAssetMetadataChanged });
+        const canvas = wrapper.instance();
+
+        canvas.editor.onRegionSelected("test1", null);
+
+        const newTag = "newTag";
+        canvas.applyTag(newTag);
+
+        const original = getAssetMetadata();
+        const expected = cloneWithUpdatedRegionTags(original, "test1", [newTag]);
+
+        expect(onAssetMetadataChanged).toBeCalledWith(expected);
+        expect(wrapper.state().currentAsset.regions[0].tags).toEqual([newTag]);
+    });
+
+    it("Adds new locked tag to selected region", () => {
+        const wrapper = createComponent();
+        const onAssetMetadataChanged = jest.fn();
+        const newTag = "newTag";
+        wrapper.setProps({
+            onAssetMetadataChanged,
+            lockedTags: [newTag],
+        });
+        const canvas = wrapper.instance();
+
+        canvas.editor.onRegionSelected("test1", false);
+
+        canvas.applyTag(newTag);
+
+        const original = getAssetMetadata();
+        const expected = cloneWithUpdatedRegionTags(original, "test1", [newTag]);
+
+        expect(onAssetMetadataChanged).toBeCalledWith(expected);
+        expect(wrapper.state().currentAsset.regions[0].tags).toEqual([newTag]);
+    });
+
+    it("Removes old locked tag from selected region", () => {
```

Comment:
Is description accurate - it doesn't appear that this test is removing any tags, but applying new ones.

Reply:
You're right, the logic is a bit off. I meant to apply one of the locked tags and have it missing from the lockedTags array, simulating the action of having regions selected and deslecting a locked tag. I'll make the fix